syntax = "proto3";
// import "google/protobuf/timestamp.proto";

package policestation;

service policestationService {
  rpc Police_station(PolicestationRequest) returns (PolicestationResponse);
  rpc InsertCharges(InsertChargesRequest ) returns (InsertChargesResponse);
  rpc DeleteCharges(DeleteChargesRequest) returns (DeleteChargesResponse);
  rpc FetchNTSA(FetchNTSARequest) returns (FetchNTSAResponse);
  rpc FetchCharges(FetchchargesRequest) returns (FetchchargesResponse);
 

}

message PolicestationRequest {
  int32 Police_station_code =1;
}

message PolicestationResponse{
  repeated Station_entries  station_entries = 1;


}
message Station_entries {
  // Define the fields for each data entry
  // This will depend on the structure of your MySQL data
  // For example, if you have a table with columns "id", "name", and "age",
  // you can define the fields as follows:
  string station_name =1;
  int32 Police_station_code =2;

}

message InsertChargesRequest {
  string Number_plate = 1;
  string charges = 2;
  int32 Police_station_code_id = 3;
}


message InsertChargesResponse {
  bool success = 1;
  string error = 2;
}


 

message FetchNTSARequest {
   string Number_plate =1;
}

message FetchNTSAResponse{
    
    repeated DataNTSAEntry data_entries = 1;

}

 
message DataNTSAEntry {
    // Define the fields for each data entry
    // This will depend on the structure of your MySQL data
    // For example, if you have a table with columns "id", "name", and "age",
    // you can define the fields as follows:
    string Number_plate = 1;
    string email = 2;
    int32 id = 3;
    string date = 4;
}



message DeleteChargesRequest {
    int32 ID = 1;
    int32 police_code =2;
 
}
 
message DeleteChargesResponse {
   bool success = 1;
   string error = 2;
 
}


message FetchchargesRequest {
   int32 Police_station_code_id =1;
}

message FetchchargesResponse{
    
    repeated Charges_entries  Charges_entries = 1;

}

 
message Charges_entries {
    // Define the fields for each data entry
    // This will depend on the structure of your MySQL data
    // For example, if you have a table with columns "id", "name", and "age",
    // you can define the fields as follows:
    int32 id =1;
    string Number_plate = 2;
    string charges = 3;
    string date = 4;
    

}



